src/zenml/integrations/aws/container_registries/aws_container_registry.py

```diff
@@ -70,9 +70,27 @@ def prepare_image_push(self, image_name: str) -> None:
         Raises:
             ValueError: If the docker image name is invalid.
         """
-        response = boto3.client(
-            "ecr", region_name=self._get_region()
-        ).describe_repositories()
+        # Find repository name from image name
+        match = re.search(f"{self.config.uri}/(.*):.*", image_name)
+        if not match:
+            raise ValueError(f"Invalid docker image name '{image_name}'.")
+        repo_name = match.group(1)
+
+        try:
+            response = boto3.client(
+                "ecr", region_name=self._get_region()
+            ).describe_repositories()
+        except NoCredentialsError:
+            logger.warning(
+                "Amazon ECR requires you to create a repository before you can "
+                f"push an image to it. ZenML is trying to push the image "
+                f"{image_name} but could not find any repositories because "
+                "your local AWS credentials are not set. We will try to push "
+                "anyway, but in case it fails you need to create a repository "
+                f"named `{repo_name}`."
+            )
+            return
```

Comment:
Do we need this `return` if it's going to continue onwards to run the push anyway?

Reply:
or even do we want it?

Reply:
This function doesn't actually do the push, it just checks what kind repositories exist and raises a warning if it does not find the repository it expects. Thus, the `return` simply prevents the other checks (since we have already raised a warning and don't want to raise another back to back)